### ISSUE TITLE
copy cmd has no lock-file

### ISSUE THREAD

**Issue by chymian** [creativeprojects/resticprofile#97]
during a copy run, there is no lock-file generated for that repo. is that on purpose?
which could maybe lead to a backup & prune operation while a copy is already running. that seems like  a potential for corrupting the remote repo?

also there needs to be a `schedule` entry in the copy-section of the profile that the cmd. works at all.

**Comment by creativeprojects** [comment]
1. Do you mean a _resticprofile_ lock file or a _restic_ lock file?

As far as _resticprofile_ is concerned, there's no separate lock file for the second repository, but the profile lock file is still used.

```yaml
profile:
    repository: "/backup/original"
    password-file: key
    lock: "/backup/resticprofile.lock"
    copy:
        repository: "/backup/copy"
        password-file: other_key
```

In such a configuration, the lock file `/backup/resticprofile.lock` will be used for any command in the profile (`snapshots`, `backup`, `check`, `copy`, etc.)

2. I don't understand what you're asking about `schedule`: yes there's a `schedule` section on the `copy` command, if it's what you're asking

**Comment by chymian** [comment]
>  Do you mean a resticprofile lock file or a restic lock file?

resticprofile-lock. the repo it's self get locked by restic, as I found out.

the lock-file is defined in a base-statement and inherited to backup, check and copy sections.
backup & check create a lockfile, copy does not.

```
# Variables
{{ $repo_local := "/dpool/backup/restic/sagar" }}
# repo access via rclone setup to avoid cleartext password here
{{ $repo_s3 := "rclone:s3:/sagar/restic/" }}



# nested template declarations
{{ define "base_backup" }}
    exclude = [ "/var/log/{{ .Profile.Name }}-backup.log" ]
    exclude-file = [
        "{{ .ConfigDir }}/global.excludes",
        "{{ .ConfigDir }}/{{ .Profile.Name }}.excludes"
    ]
    check-before = false
    schedule-lock-wait = "10m"
    exclude-caches = true
    no-error-on-warning = true
    schedule-permission = "system"
    tag = [ "{{ .Profile.Name }}" ]
{{ end }}

[base]
initialize = true
lock = "/tmp/resticprofile-{{ .Profile.Name }}.lock"
force-inactive-lock = true
run-after-fail = "/usr/local/bin/resticprofile-send-error.sh  root"


  [base.backup]
  host = true

  [base.retention]
  after-backup = true
  before-backup = false
  compact = false
  #keep-last = 3
  keep-within = "3h"
  keep-tag = [ "forever" ]
  prune = true
  host = true

  [base.check]
  read-data = true
  schedule-permission = "system"

  [base.copy]
  repository = "{{ $repo_s3 }}/{{ .Profile.Name }}"
  password-file2 = "key"

[repos]
inherit = "base"
password-file = "key"
option = [
    "rclone.program=/usr/bin/rclone",
    "rclone.args=serve restic --stdio --transfers 10 --verbose"
]


[root]
repository = "{{ $repo_local }}/{{ .Profile.Name }}"
inherit = "repos"

  [root.backup]
  source = [ "/", "/boot", "/boot/efi" ]
  {{ template "base_backup" . }}
  schedule = "11/3:11"
  one-file-system = true

  [root.retention]
  keep-daily = 7
  keep-weekly = 4
  keep-monthly = 3
  keep-yearly = 0


  [root.copy]
  schedule = "daily"

  [root.check]
  # if scheduled, will check the repository the first day & 13th of each month at 13:44am
  schedule = "*-*-01,13 13:44"
....
```
> I don't understand what you're asking about schedule: yes there's a schedule section on the copy command, if it's what you're asking

at the moment I cannot reproduce, because of [issue 96](https://github.com/creativeprojects/resticprofile/issues/96). but while developing the config, I had a point, when there was no schedule-line defined in the copy-section, `copy` did not work.
please postpone, I will recheck when [issue 96](https://github.com/creativeprojects/resticprofile/issues/96) is fixed.

**Comment by creativeprojects** [comment]
> even so, there are parameters inherited, from root.retenion, without a vaild entry in the section, it's getting ignored.

oh wow that's an interesting defect, I never noticed that before 🤔 I'll look into it

**Comment by creativeprojects** [comment]
I was getting worried about this, but it turns out it is a display bug only for the `profiles` command.

If you use the `show` command, it should display the full resulting profile properly: `resticprofile -n s3 show`

I'll fix the display on the `profiles` command 👍🏻

**Comment by creativeprojects** [comment]
Fixed in v0.17.0 release 👍🏻
